src/nexusrpc/handler/_core.py

```diff
@@ -0,0 +1,458 @@
+from __future__ import annotations
+
+import asyncio
+import concurrent.futures
+import inspect
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Optional,
+    Sequence,
+    Union,
+)
+
+from typing_extensions import Self
+
+import nexusrpc
+import nexusrpc._service
+from nexusrpc.handler._util import is_async_callable
+
+from .. import OperationInfo
+from .._serializer import LazyValue
+from ._common import (
+    CancelOperationContext,
+    FetchOperationInfoContext,
+    FetchOperationResultContext,
+    HandlerError,
+    HandlerErrorType,
+    StartOperationContext,
+    StartOperationResultAsync,
+    StartOperationResultSync,
+)
+from ._operation_handler import (
+    OperationHandler,
+    collect_operation_handler_factories,
+)
+
+# TODO(preview): show what it looks like to manually build a service implementation at runtime
+# where the operations may be based on some runtime information.
+
+# TODO(preview): pass pyright strict mode "python.analysis.typeCheckingMode": "strict"
+# TODO(preview): pass mypy
+
+
+class BaseHandler(ABC):
+    """
+    A Nexus handler, managing a collection of Nexus service handlers.
+
+    Operation requests are delegated to a :py:class:`ServiceHandler` based on the service
+    name in the operation context.
+    """
+
+    def __init__(
+        self,
+        user_service_handlers: Sequence[Any],
```

Comment:
If this is a base handler, why accept these? What if I want to implement my base handler without any user service handlers? I think we should have a true abstract handler with no implementation defaults (i.e. basically an interface). Can we move all of this constructor stuff to the `Handler` class and get rid of this constructor? If we need some kind of shared utilities between `Handler` and `SyncHandler` so they can accept similar constructor arguments, that can make sense (or even a shared base class), but IMO we still need a no-impl/no-constructor interface for handlers.

Reply:
Done, `nexusrpc.handler` now looks like

<img width="462" alt="image" src="https://github.com/user-attachments/assets/f1b9c56b-6fd5-4bd4-ae60-5c710777ba22" />

while `syncio.handler` looks like

<img width="390" alt="image" src="https://github.com/user-attachments/assets/be1a5f6a-1927-4938-84f3-701039b5ad49" />